cmake/nokia_ngage.cmake

```diff
@@ -0,0 +1,133 @@
+  "${SDL3_SOURCE_DIR}/src/*.c"
+  "${SDL3_SOURCE_DIR}/src/atomic/*.c"
+  "${SDL3_SOURCE_DIR}/src/audio/*.c"
+  "${SDL3_SOURCE_DIR}/src/audio/ngage/*.c"
+  "${SDL3_SOURCE_DIR}/src/audio/ngage/*.cpp"
+  "${SDL3_SOURCE_DIR}/src/core/*.c"
+  "${SDL3_SOURCE_DIR}/src/core/ngage/*.c"
+  "${SDL3_SOURCE_DIR}/src/core/ngage/*.cpp"
+  "${SDL3_SOURCE_DIR}/src/cpuinfo/*.c"
+  "${SDL3_SOURCE_DIR}/src/dynapi/*.c"
+  "${SDL3_SOURCE_DIR}/src/events/*.c"
+  "${SDL3_SOURCE_DIR}/src/file/*.c"
+  "${SDL3_SOURCE_DIR}/src/file/generic/*.c"
+  "${SDL3_SOURCE_DIR}/src/filesystem/*.c"
+  "${SDL3_SOURCE_DIR}/src/filesystem/ngage/*.c"
+  "${SDL3_SOURCE_DIR}/src/filesystem/ngage/*.cpp"
+  "${SDL3_SOURCE_DIR}/src/filesystem/posix/*.c"
+  "${SDL3_SOURCE_DIR}/src/hidapi/*.c"
+  "${SDL3_SOURCE_DIR}/src/io/*.c"
+  "${SDL3_SOURCE_DIR}/src/io/generic/*.c"
+  "${SDL3_SOURCE_DIR}/src/libm/s_isinff.c"
+  "${SDL3_SOURCE_DIR}/src/libm/s_isnanf.c"
+  "${SDL3_SOURCE_DIR}/src/locale/*.c"
+  "${SDL3_SOURCE_DIR}/src/locale/ngage/*.cpp"
+  "${SDL3_SOURCE_DIR}/src/main/*.c"
+  "${SDL3_SOURCE_DIR}/src/main/ngage/*.c"
+  "${SDL3_SOURCE_DIR}/src/main/ngage/*.cpp"
+  "${SDL3_SOURCE_DIR}/src/misc/*.c"
+  "${SDL3_SOURCE_DIR}/src/misc/dummy/*.c"
+  "${SDL3_SOURCE_DIR}/src/power/*.c"
+  "${SDL3_SOURCE_DIR}/src/render/*.c"
+  "${SDL3_SOURCE_DIR}/src/render/ngage/*.c"
+  "${SDL3_SOURCE_DIR}/src/render/ngage/*.cpp"
+  "${SDL3_SOURCE_DIR}/src/stdlib/*.c"
+  "${SDL3_SOURCE_DIR}/src/storage/*.c"
+  "${SDL3_SOURCE_DIR}/src/storage/generic/*.c"
+  "${SDL3_SOURCE_DIR}/src/thread/*.c"
+  "${SDL3_SOURCE_DIR}/src/thread/generic/*.c"
+  "${SDL3_SOURCE_DIR}/src/time/*.c"
+  "${SDL3_SOURCE_DIR}/src/time/ngage/*.cpp"
+  "${SDL3_SOURCE_DIR}/src/time/unix/*.c"
+  "${SDL3_SOURCE_DIR}/src/timer/*.c"
+  "${SDL3_SOURCE_DIR}/src/timer/ngage/*.cpp"
+  "${SDL3_SOURCE_DIR}/src/tray/*.c"
+  "${SDL3_SOURCE_DIR}/src/tray/dummy/*.c"
+  "${SDL3_SOURCE_DIR}/src/video/*.c"
+  "${SDL3_SOURCE_DIR}/src/video/ngage/*.c"
+  "${SDL3_SOURCE_DIR}/src/video/yuv2rgb/*.c")
+
+add_library(${PROJECT_NAME} STATIC ${SDL3_sources})
+
+target_include_directories(
+  ${PROJECT_NAME}
+  PUBLIC
+  "${SDL3_SOURCE_DIR}/include"
+  PRIVATE
+  "${SDL3_SOURCE_DIR}/src"
+  "${SDL3_SOURCE_DIR}/include/build_config")
+
+target_compile_definitions(
+  ${PROJECT_NAME}
+  PUBLIC
+  SDL_STATIC_LIB
+  ${GCC_MODULE_DEFS})
+
+set(test_static_libs
+  ${CMAKE_CURRENT_BINARY_DIR}/libSDL3.a)
+
+set(test_libs
+  ${CMAKE_CURRENT_BINARY_DIR}/libSDL3.a
+  ${EPOC_LIB}/NRenderer.lib
+  ${EPOC_LIB}/3dtypes.a
+  ${EPOC_LIB}/cone.lib
+  ${EPOC_PLATFORM}/gcc/lib/gcc-lib/arm-epoc-pe/2.9-psion-98r2/libgcc.a
+  ${EPOC_PLATFORM}/ngagesdk/lib/gcc/arm-epoc-pe/4.6.4/libgcc_ngage.a
+  ${EPOC_LIB}/mediaclientaudiostream.lib
+  ${EPOC_LIB}/charconv.lib
+  ${EPOC_LIB}/bitgdi.lib
+  ${EPOC_LIB}/euser.lib
+  ${EPOC_LIB}/estlib.lib
+  ${EPOC_LIB}/ws32.lib
+  ${EPOC_LIB}/hal.lib
+  ${EPOC_LIB}/fbscli.lib
+  ${EPOC_LIB}/efsrv.lib
+  ${EPOC_LIB}/scdv.lib
+  ${EPOC_LIB}/gdi.lib)
```

Comment:
Add `${EPOC_LIB}`, `${EPOC_PLATFORM}/gcc/lib/gcc-lib/arm-epoc-pe/2.9-psion-982` and `${EPOC_PLATFORM}/ngagesdk/lib/gcc/arm-epoc-pe/4.6.4/` to [CMAKE_<LANG>_STANDARD_LINK_DIRECTORIES](https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_STANDARD_LINK_DIRECTORIES.html) in your CMake toolchain file.
(`CMAKE_C_STANDARD_LINK_DIRECTORIES` and `CMAKE_CXX_STANDARD_LINK_DIRECTORIES`)

Reply:
This command was added in CMake version 3.31. Even the latest preview of Visual Studio 2022 includes only 3.30.5. Any alternative?

Reply:
I had a closer look. Since these paths are also required by the custom-defined linking procedure, we can't use this mechanism either way.